azure-pipelines.yml

```diff
@@ -51,6 +51,16 @@ steps:
   workingDirectory: $(Pipeline.Workspace)/target/python-wheels/
   displayName: 'Install Python dependencies'
 
+- script: |
+    set -ex
+    # Install .NET CORE
+    curl -sSL https://packages.microsoft.com/keys/microsoft.asc | sudo apt-key add -
+    sudo apt-add-repository https://packages.microsoft.com/debian/10/prod
+    sudo apt-get update
+    sudo apt-get -y install -f
```

Comment:
This now works, but I don't fully understand why we need this line immediately after the `sudo apt-get update` call.

Reply:
@jleveque yea, I am checking.

Reply:
I think some libs are required by .NET:
```
Dependencies
When you install with a package manager, these libraries are installed for you. But, if you manually install .NET Core or you publish a self-contained app, you'll need to make sure these libraries are installed:

libc6
libgcc1
libgssapi-krb5-2
libicu52 (for 8.x)
libicu57 (for 9.x)
libicu63 (for 10.x)
libicu67 (for 11.x)
libssl1.0.0 (for 8.x)
libssl1.1 (for 9.x-11.x)
libstdc++6
zlib1g
```

Reply:
But I don't see any of those packages mentioned in the output of `sudo apt-get -y install -f`. Plus, this command is usually used after a package install fails due to missing dependencies. I've never needed to run it immediately after adding a new repository.

Reply:
yea, strange, i have do some test in https://github.com/Azure/sonic-snmpagent/pull/215 try to install dependent libs, but no success.

Reply:
after many testing, 'sudo apt-get -y install -f' can and only can be set at this place.

Reply:
Understood, but do you understand _why_ it is necessary? What is it fixing?

Reply:
As Luoqi mentioned, -f fix below:
```
I compared the log with previous one, and the previous one seems suspicious

2021-06-29T03:04:13.8494132Z The following additional packages will be installed:
2021-06-29T03:04:13.8499448Z   libhiredis0.14
2021-06-29T03:04:13.8986264Z The following packages will be REMOVED:
2021-06-29T03:04:13.8993644Z   libibverbs-dev libnl-3-dev libnl-route-3-dev libopenmpi-dev
2021-06-29T03:04:13.9005991Z The following packages will be upgraded:
2021-06-29T03:04:13.9011224Z   libhiredis0.14
2021-06-29T03:04:13.9242794Z 1 upgraded, 0 newly installed, 4 to remove and 2 not upgraded.
The new apt source messed up the local package database.
```
and i try to do some experimenents, but all not work:
(snmpagent repo)
```
put add source at begin:
https://dev.azure.com/mssonic/build/_build/results?buildId=21862&view=logs&j=83516c17-6666-5250-abde-63983ce72a49&t=c10d5f44-55ce-55d7-7975-407ed75d9a96
remove source and only wget debs:
https://dev.azure.com/mssonic/build/_build/results?buildId=21866&view=logs&j=83516c17-6666-5250-abde-63983ce72a49&t=c10d5f44-55ce-55d7-7975-407ed75d9a96
try dotnet-install.sh, no error but no color bar:
https://dev.azure.com/mssonic/build/_build/results?buildId=21868&view=logs&j=83516c17-6666-5250-abde-63983ce72a49&t=c10d5f44-55ce-55d7-7975-407ed75d9a96
```